src/smdba/postgresqlgate.py

```diff
@@ -188,19 +191,23 @@ def check(self):
         if os.popen('/usr/bin/postmaster --version').read().strip().split(' ')[-1] < '9.1':
             raise GateException("Core component is too old version.")
         elif not os.path.exists("/etc/sysconfig/postgresql"):
-            raise GateException("Custom core component? Please strictly use SUSE components only!")
+            raise GateException("Custom database component? Please strictly use SUSE components only!")
         elif not os.path.exists("/usr/bin/psql"):
             msg = 'operations'
         elif not os.path.exists("/usr/bin/postmaster"):
-            msg = 'core'
+            msg = 'database'
         elif not os.path.exists("/usr/bin/pg_ctl"):
             msg = 'control'
         elif not os.path.exists("/usr/bin/pg_basebackup"):
             msg = 'backup'
-
         if msg:
             raise GateException("Cannot find required %s component." % msg)
 
+        # Prevent running this tool within the PostgreSQL data directory
+        # See bsc#1024058 for details
+        if os.path.abspath(".") == self.config["pcnf_pg_data"]:
+            raise GateException("Please do not call SMDBA inside the '{0}' directory.".format(os.path.abspath(".")))
+
```

Comment:
Seems my comment was marked as outdated. So I repeat it:-)

I see problems with this exact compare. What happens if:
- self.config["pcnf_pg_data"] is sometimes configured with a final / and and sometimes without?
- what if I am in a subdir of self.config["pcnf_pg_data"] like /var/lib/pgsql/data/pg_log/ ?

I think we need to check if we are somewhere inside that dir.